src/services/GroupMeService.ts

```diff
@@ -37,8 +40,93 @@ export class GroupMeService implements MessagingService {
     console.log("response done");
     return right(0);
   }
+
+  /**
+   * Get list of users the bot is registered to according to the env
+   */
+  async getUsers(): Promise<Either<Error, GroupMeUser[]>> {
+    try {
+      const { members } = await this.getGroupInfo();
+      return right(members);
+    } catch (e) {
+      return left(e as Error);
+    }
+  }
+
+  /**
+   * Get group info
+   */
+  async getGroupInfo(): Promise</*Either<Error, */ GroupMeGroupInfo> /*> */ {
+    // try {
+    const { response } = await got
+      .get(`${GroupMeAPI}/groups/${this.envConfigService.GroupID}?token=${this.envConfigService.GroupMeAPIKey}`)
+      .json();
+    return response as GroupMeGroupInfo;
+    // return right(response as GroupMeGroupInfo);
+    // } catch (e) {
+    //   return left(e as Error);
+    // }
+  }
 }
 
+export type GroupMeUser = {
+  user_id: string;
+  nickname: string;
+  image_url: string;
+  id: string;
+  muted: boolean;
+  autokicked: boolean;
+  roles: string[];
+  name: string;
+};
+
+export type GroupMeGroupInfo = {
+  id: string;
+  group_id: string;
+  name: string;
+  phone_number: string;
+  type: string;
+  description: string;
+  image_url: null;
+  creator_user_id: string;
+  created_at: number;
+  updated_at: number;
+  muted_until: null; // not sure about this
+  office_mode: boolean;
+  share_url: string;
+  share_qr_code_url: string;
+  members: GroupMeUser[];
+  messages: {
+    count: number;
+    last_message_id: string;
+    last_message_created_at: number;
+    preview: {
+      nickname: string;
+      text: string;
+      image_url: string;
+      attachments: GroupMeAttachment[];
+    };
+  };
+  max_members: number;
+};
+
+export type GroupMeMention = { loci: Array<[number, number]>; type: "mentions"; user_ids: string[] };
```

Comment:
Is it good to "hardcode" string type like mentions here? Should to it to image and location below. Maybe they should be constants? Add helper constructor funcs?

Reply:
fine to me